crates/iceberg/src/expr/mod.rs

```diff
@@ -18,25 +18,129 @@
 //! This module contains expressions.
 
 mod term;
+
+use std::fmt::{Display, Formatter};
+
 pub use term::*;
 mod predicate;
 pub use predicate::*;
 
 /// Predicate operators used in expressions.
+///
+/// The discriminant of this enum is used for determining the type of the operator, see
+/// [`PredicateOperator::is_unary`], [`PredicateOperator::is_binary`], [`PredicateOperator::is_set`]
 #[allow(missing_docs)]
+#[derive(Debug, Clone, Copy)]
+#[repr(u16)]
 pub enum PredicateOperator {
```

Comment:
Good suggestion. I think the `AlwaysTrue`/`AlwaysFalse` would be quite useful, but I prefer to add them in later prs since this pr is already a little too large for review.

Reply:
Tracked in #224

Reply:
Sounds good, thanks for creating the issue 👍